vaadin-cdi-itest/src/main/java/com/vaadin/cdi/itest/regression/RemoveOldContentView.java

```diff
@@ -0,0 +1,96 @@
+import com.vaadin.flow.router.RouterLayout;
+
+@UIScoped
+@Route("")
+public class RemoveOldContentView extends Div
+        implements RouterLayout {
+
+    public static String NAVIGATE_TO_ANOTHER_ROUTE_INSIDE_MAIN_LAYOUT_BUTTON_ID =
+            "navigate-to-another-route-inside-main-layout-button-id";
+    public static String NAVIGATE_TO_ANOTHER_ROUTE_OUTSIDE_MAIN_LAYOUT_BUTTON_ID = "navigate-to-another-layout-button-id";
+    public static String NAVIGATE_BACK_FROM_ANOTHER_ROUTE_INSIDE_MAIN_LAYOUT_BUTTON_ID = "navigate-back-from-another-route-inside-main-layout-button-id";
+    public static String NAVIGATE_BACK_FROM_ANOTHER_ROUTE_OUTSIDE_MAIN_LAYOUT_BUTTON_ID = "navigate-back-from-another-route-outside-main-layout-button-id";
+    public static String SUB_LAYOUT_ID = "sub-layout-id";
+
+    public RemoveOldContentView() {
+        add(new Span("This is a topmost parent router layout"));
+    }
+
+    @UIScoped
+    @ParentLayout(RemoveOldContentView.class)
+    public static class SubLayout extends Div implements RouterLayout {
+        public SubLayout() {
+            setId(SUB_LAYOUT_ID);
+            add(new Span("This is a sub router layout"));
+        }
+    }
+
+    @Route(value = "first-child-route", layout = SubLayout.class)
+    public static class FirstView extends Div {
+
+        public FirstView() {
+            add(new Span("This is a child route inside main layout"));
+            NativeButton navigateToAnotherViewButton = new NativeButton(
+                    "Navigate to another route inside Main Layout",
+                    click -> UI.getCurrent().navigate(SecondView.class));
+            navigateToAnotherViewButton.setId(
+                    NAVIGATE_TO_ANOTHER_ROUTE_INSIDE_MAIN_LAYOUT_BUTTON_ID);
+            add(navigateToAnotherViewButton);
+            NativeButton navigateToAnotherLayoutButton = new NativeButton(
+                    "Navigate to another route outside Main Layout",
+                    click -> UI.getCurrent().navigate(
+                            CustomUIScopeAnotherLayout.ThirdView.class));
+            navigateToAnotherLayoutButton.setId(
+                    NAVIGATE_TO_ANOTHER_ROUTE_OUTSIDE_MAIN_LAYOUT_BUTTON_ID);
+            add(navigateToAnotherLayoutButton);
+        }
+    }
+
+    @Route(value = "second-child-route", layout = RemoveOldContentView.class)
+    public static class SecondView extends Div {
+
+        public SecondView() {
+            add(new Span("This is another route inside Main Layout"));
+            NativeButton navigateToChildView = new NativeButton(
+                    "Navigate to first route",
+                    click -> UI.getCurrent().navigate(FirstView.class));
+            navigateToChildView.setId(
+                    NAVIGATE_BACK_FROM_ANOTHER_ROUTE_OUTSIDE_MAIN_LAYOUT_BUTTON_ID);
+            add(navigateToChildView);
+        }
+    }
+
+    @Route("secondary")
+    public static class CustomUIScopeAnotherLayout extends Div
```

Comment:
It's not UI scoped.
And looks like it should not be.
So I suggest to change the name,

Reply:
Thanks! It indeed wrong. Renamed.